File: activemq-broker/src/main/java/org/apache/activemq/broker/BrokerService.java
/**
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.apache.activemq.broker;

import java.net.URI;
import java.util.List;

import org.apache.activemq.Service;
import org.apache.activemq.broker.region.RegionBroker;
import org.apache.activemq.usage.SystemUsage;

public class BrokerService implements Service {

    public static final String DEFAULT_PORT = "61616";
    public static final String DEFAULT_BROKER_NAME = "localhost";

	public Object getVmConnectorURI() {
        // TODO Auto-generated method stub
        return null;
    }

    public RegionBroker getRegionBroker() {
        // TODO Auto-generated method stub
        return null;
    }

    public boolean isUseJmx() {
        // TODO Auto-generated method stub
        return false;
    }

    public ManagementContext getManagementContext() {
        // TODO Auto-generated method stub
        return null;
    }

    public <T> T addNetworkConnector(URI connectorURI) {
        // TODO Auto-generated method stub
        return null;
    }

    public String getBrokerName() {
        // TODO Auto-generated method stub
        return null;
    }

    public Broker getBroker() {
        return null;
    }

    public SslContext getSslContext() {
        // TODO Auto-generated method stub
        return null;
    }

	public void start() throws Exception {
		// TODO Auto-generated method stub
		
	}

	public void stop() throws Exception {
		// TODO Auto-generated method stub
		
	}

	public void setPersistent(boolean b) {
		// TODO Auto-generated method stub
		
	}

	public TransportConnector addConnector(String uri) {
		// TODO Auto-generated method stub
		return null;
	}

	public void setUseJmx(boolean b) {
		// TODO Auto-generated method stub
		
	}

	public <T> void setDestinations(T[] activeMQDestinations) {
		// TODO Auto-generated method stub
		
	}

	public void addConnector(org.apache.activemq.broker.TransportConnector connector) {
		// TODO Auto-generated method stub
		
	}

	public List<TransportConnector> getTransportConnectors() {
		return null;
	}

	public void setDestinationPolicy(org.apache.activemq.broker.region.policy.PolicyMap policyMap) {
		// TODO Auto-generated method stub
		
	}

	public SystemUsage getSystemUsage() {
		// TODO Auto-generated method stub
		return null;
	}

	public <T> T getAdminView() {
		// TODO Auto-generated method stub
		return null;
	}

    public void addShutdownHook(Runnable runnable) {
		// TODO Auto-generated method stub
		
	}

	public boolean isStarted() {
		// TODO Auto-generated method stub
		return false;
	}


}

File: apollo-itests/src/test/scala/org/apache/activemq/apollo/BrokerService.scala
/**
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.apache.activemq.apollo

import broker.{BrokerFactory, Broker}
import javax.jms.ConnectionFactory
import java.net.InetSocketAddress
import util.{Logging, ServiceControl}
import java.util.Hashtable
import javax.naming.InitialContext


/**
 *
 */
trait BrokerService extends Logging {

  var broker: Broker = null
  var port = 0
  var started = false

  def start = {
    try {
      info("Loading broker configuration from the classpath with URI: " + broker_config_uri)
      broker = BrokerFactory.createBroker(broker_config_uri)
      ServiceControl.start(broker, "Starting broker")
      port = broker.get_socket_address.asInstanceOf[InetSocketAddress].getPort
    }
    catch {
      case e:Throwable => e.printStackTrace
      throw e
    }
  }


  def stop = ServiceControl.stop(broker, "Stopping broker")

  def broker_config_uri:String

  def getConnectionFactory = {
    if (!started) {
      start
    }
    val jndiConfig = new Hashtable[String, String]
    jndiConfig.put("java.naming.factory.initial", getInitialContextFactoryClass)
    jndiConfig.put("java.naming.provider.url", getConnectionUri)
    jndiConfig.put("java.naming.security.principal", "admin")
    jndiConfig.put("java.naming.security.credentials", "password")
    val ctx = new InitialContext(jndiConfig)
    ctx.lookup("ConnectionFactory").asInstanceOf[ConnectionFactory]
  }

  protected def getInitialContextFactoryClass:String

  def getConnectionUri:String
}

/**
 *
 */
class StompBroker extends BrokerService {

  def broker_config_uri = "xml:classpath:apollo-stomp.xml"

  protected def getInitialContextFactoryClass = "org.fusesource.stompjms.jndi.StompJmsInitialContextFactory"

  def getConnectionUri = "tcp://localhost:%s".format(port);

}

/**
 *
 */
class OpenwireBroker extends BrokerService {

  def broker_config_uri = "xml:classpath:apollo-openwire.xml"

  protected def getInitialContextFactoryClass = "org.apache.activemq.jndi.ActiveMQInitialContextFactory"

  def getConnectionUri = "tcp://localhost:%s".format(port)

}
